activesupport/lib/active_support/core_ext/module/attribute_accessors.rb

```diff
@@ -203,4 +203,24 @@ def mattr_accessor(*syms, instance_reader: true, instance_writer: true, instance
     mattr_writer(*syms, instance_writer: instance_writer, instance_accessor: instance_accessor, default: default, location: location)
   end
   alias :cattr_accessor :mattr_accessor
+
+  def set(**attributes)
```

Comment:
As far as I understood this just restores the previous atributes values after the block. To me `set` sounds like a permanent change. How about `with`? Is it already taken?

An example from above:

```ruby
def with_service(name, &block)
  ActiveStorage::Blob.with service: name.nil? ? nil : ActiveStorage::Blob.services.fetch(name), &block
end
```

```suggestion
  def with(**attributes)
```